src/components/Notification/Notification.js

```diff
@@ -486,34 +508,10 @@ export class InlineNotification extends Component {
       className
     );
 
-    const NotificationIcon = kind => {
-      if (componentsX) {
-        const NotificationIconX = this.useIcon(kind);
-        return (
-          <NotificationIconX
-            className={`${prefix}--inline-notification__icon`}
-          />
-        );
-      }
-      switch (kind) {
-        case 'warning':
-          return a11yIconWarningSolid(prefix, notificationType);
-        default:
-          return (
-            <Icon
-              description={this.props.iconDescription}
-              className={`${prefix}--inline-notification__icon`}
-              aria-label="close"
-              icon={this.useIcon(kind)}
-            />
-          );
-      }
-    };
-
     return (
       <div {...other} role={role} kind={kind} className={classes}>
         <div className={`${prefix}--inline-notification__details`}>
-          {NotificationIcon(kind)}
+          {NotificationIcon({ notificationType, kind, iconDescription })}
```

Comment:
And from here I see you turned `NotificationIcon()` function to a React component, without doing it on-purpose 😉